common/persistence/visibility/store/query/converter_legacy.go

```diff
@@ -451,9 +456,20 @@ func (c *comparisonExprConverter) Convert(expr sqlparser.Expr) (elastic.Query, e
 		return nil, NewConverterError("operator '%v' not allowed in comparison expression", comparisonExpr.Operator)
 	}
 
-	tp, err := c.saNameType.GetType(colName)
-	if err != nil {
-		return nil, err
+	// Get field type, checking CHASM mapper first if applicable
+	var tp enumspb.IndexedValueType
+	if sadefs.IsChasmSearchAttribute(colName) {
+		var err error
```

Comment:
`err` is already defined in the scope of this function, no need to create a new var.